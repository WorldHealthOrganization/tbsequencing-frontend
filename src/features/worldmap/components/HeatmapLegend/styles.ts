import { css } from '@emotion/react';
import appColors from '../../../../styles/colors';

const BASE_STYLE = {
  display: 'inline-block', // FIX FOR SAFARI
};

const MARGIN_STYLE = {
  ...BASE_STYLE,
  marginLeft: 8,
};

export const container = css({
  display: 'flex',
});

export const white = css({
  ...BASE_STYLE,
  border: `1px solid ${appColors.primary.navy}`,
  backgroundColor: appColors.neutral.white,
});

export const blue = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoBlue.light,
});

export const lightBlue = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoBlue.lighter,
});

export const lighterBlue = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoBlue.lightest,
});

export const lightestOrange = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoOrange.lightest,
});

export const lighterOrange = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoOrange.lighter,
});

export const lightOrange = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.tints.whoOrange.light,
});

export const orange = css({
  ...MARGIN_STYLE,
  backgroundColor: appColors.secondary.whoOrange,
});
