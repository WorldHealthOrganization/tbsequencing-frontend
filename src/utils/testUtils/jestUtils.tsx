import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
