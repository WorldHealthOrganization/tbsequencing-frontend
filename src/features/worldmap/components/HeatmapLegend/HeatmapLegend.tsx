/** @jsxImportSource @emotion/react */
import React from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import MapLegendSquare from '../MapLegendSquare';

const legendItemStyles: SerializedStyles[] = [
  styles.white,
  styles.blue,
  styles.lightBlue,
  styles.lighterBlue,
  styles.lightestOrange,
  styles.lightestOrange,
  styles.lighterOrange,
  styles.lightOrange,
  styles.orange,
];

export const HeatmapLegend = () => (
  <legend css={styles.container}>
    {
      // in this case we can use index, since array will never be reordered and it is static
      /* eslint-disable-next-line react/no-array-index-key */}
    {legendItemStyles.map((style, index) => <MapLegendSquare key={`${style.name}-${index}`} style={style} />)}
  </legend>
);

export default HeatmapLegend;
