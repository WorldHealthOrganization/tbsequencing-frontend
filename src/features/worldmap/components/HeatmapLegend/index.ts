import HeatmapLegend from './HeatmapLegend';

export default HeatmapLegend;
