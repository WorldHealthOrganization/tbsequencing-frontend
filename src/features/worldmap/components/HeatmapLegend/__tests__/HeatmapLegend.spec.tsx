import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../../../utils/testUtils/jestUtils';
import HeatmapLegend from '../HeatmapLegend';

describe('HeatmapLegend component test', () => {
  test('renders HeatmapLegend component without error', () => {
    render(
      <HeatmapLegend />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <HeatmapLegend />,
    );

    expect(tree).toMatchSnapshot();
  });
});
