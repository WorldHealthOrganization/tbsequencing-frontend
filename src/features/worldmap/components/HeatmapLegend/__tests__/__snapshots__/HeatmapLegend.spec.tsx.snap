// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeatmapLegend component test Should render correctly 1`] = `
<legend
  className="css-1pzxlvy-HeatmapLegend"
>
  <div
    className="css-1suqesg-MapLegendSquare"
  />
  <div
    className="css-lsl4m3-MapLegendSquare"
  />
  <div
    className="css-1rqy87q-MapLegendSquare"
  />
  <div
    className="css-1vpoz8s-MapLegendSquare"
  />
  <div
    className="css-1k62fbu-MapLegendSquare"
  />
  <div
    className="css-1k62fbu-MapLegendSquare"
  />
  <div
    className="css-307wo-MapLegendSquare"
  />
  <div
    className="css-v0r8h2-MapLegendSquare"
  />
  <div
    className="css-10fxyx9-MapLegendSquare"
  />
</legend>
`;
